firebase-firestore/src/main/java/com/google/firebase/firestore/core/Target.java

```diff
@@ -51,6 +53,11 @@
   private final @Nullable Bound startAt;
   private final @Nullable Bound endAt;
 
+  // DNF (Disjunctive Normal Form) representation of this target's filters.
+  // Each element in this list is an AND filter of FieldFilters.
+  // The query result is the disjunction (OR) of each of these AND filters.
+  private final List<CompositeFilter> dnf;
```

Comment:
The Target is essentially the proto representation. We should not store the DNF here.

Reply:
I see. Thinking about this more, I think we should be able to push it to SQLiteIndexManager.java as that's the only place where it's really needed.